crate/utils/src/crypto/curve_25519/operation.rs

```diff
@@ -155,18 +155,29 @@ pub fn create_ed25519_key_pair(
     Ok(KeyPair::new(private_key, public_key))
 }
 
-fn create_p_curve_key_pair(
+pub fn create_approved_ecc_key_pair(
     private_key_uid: &str,
     public_key_uid: &str,
     curve_nid: Nid,
 ) -> Result<KeyPair, KmipUtilsError> {
     let curve = EcGroup::from_curve_name(curve_nid)?;
     let kmip_curve = match curve_nid {
+        // P-CURVES
         Nid::X9_62_PRIME192V1 => RecommendedCurve::P192,
```

Comment:
```suggestion
        #[cfg(not(feature = "fips'))]
        Nid::X9_62_PRIME192V1 => RecommendedCurve::P192,
```
Since P192 should not be used in fips mode.

Reply:
done in 4d5a656a2044162d9356e159bc8723c274db3497